cmd/thanos/query.go

```diff
@@ -260,15 +266,14 @@ func runQuery(
 	}
 
 	fileSDCache := cache.New()
+	// DNS provider with default resolver.
+	dnsProvider := dns.NewProvider(nil)
 
 	var (
 		stores = query.NewStoreSet(
 			logger,
 			reg,
 			func() (specs []query.StoreSpec) {
-				// Add store specs from static flags.
-				specs = append(staticSpecs)
```

Comment:
Question: `staticSpecs` doesn't be used in any where, should we remove it?

Reply:
True. And we need to removal all definitions of this variable as well (: